File: Models/MatterEnergyScheduler/src/EnergySchedulerApi/Infrastructure/MockSolarForecastService.cs
using EnergyScheduler.Api.Models;
using EnergyScheduler.Api.Interfaces;

namespace EnergyScheduler.Api.Infrastructure;

public class MockSolarForecastService : ISolarForecastService
{
    public Task<List<SolarProduction>> GetForecastAsync(DateTime date, Location location)
    {
        var productions = new List<SolarProduction>();
        for (int i = 0; i < 48; i++) // 24 hours * 2
        {
            var time = date.AddHours(i * 0.5);
            double production = 0;
            int hour = time.Hour;
            if (hour >= 6 && hour <= 18)
            {
                // Simple mock: peak at noon
                production = 3.0 * Math.Sin(Math.PI * (hour - 6) / 12);
            }
            productions.Add(new SolarProduction { Time = time, KwProduced = production });
        }
        return Task.FromResult(productions);
    }
}

File: Models/MatterEnergyScheduler/src/EnergySchedulerApi/Infrastructure/mock_solar_forecast_service.py
from typing import List
from datetime import datetime, date, timedelta
import math
from ..Models.solar_production import SolarProduction


class MockSolarForecastService:
    async def get_forecast(self, target_date: date, household) -> List[SolarProduction]:
        """Mock solar forecast based on typical Ireland solar pattern"""
        productions = []
        base_time = datetime.combine(target_date, datetime.min.time())
        
        for i in range(48):
            time = base_time + timedelta(hours=i * 0.5)
            hour = time.hour + time.minute / 60
            production = 0
            
            # Simulate solar production curve (6 AM to 6 PM peak)
            if 6 <= hour <= 18:
                production = 3.0 * math.sin(math.pi * (hour - 6) / 12)
            
            productions.append(SolarProduction(time=time, kw_produced=production))
        
        return productions